core/src/apps/common/cbor.py

```diff
@@ -316,3 +316,13 @@ def create_array_header(size: int) -> bytes:
 
 def create_map_header(size: int) -> bytes:
     return _header(_CBOR_MAP, size)
+
+
+def less_than(value1: Value, value2: Value) -> bool:
```

Comment:
I ran some tests before writing the comment and the output of your function was the same as the output of a function, which would just compare `u < v` (without the length checks). And I think I finally understand why 🤦 

The reason is that we are comparing the CBOR encoded values. And since the length of the byte array or string is included at the beginning of the encoded value, the shorter value always comes first lexicographically. `b"\x00\x00\x00"` gets encoded as "43000000" (hex) and `b"\xFF"` gets encoded as "41FF" (hex).

But please try to verify my claim before changing it. Or if my claim is obviously false and I am missing something, please try to find some values which confirm that my claim is false.

Reply:
Ah, that makes sense.

> But please try to verify my claim before changing it. Or if my claim is obviously false and I am missing something, please try to find some values which confirm that my claim is false.

I didn't find any realistic input that would break this. The only problem I can think of are values of different types, eg. `encode(h'00') == 4100`, `encode(1234) == 1904d2`. Not sure if this is or will be relevant for us.

However, another reason for leaving the `u < v` check in place is the cbor spec - they explicitly state both sorting rules, so I think it's safer to implement them both rather than hoping that the first one is really redundant for all our present and future use cases.

Reply:
Ok, let's keep it then. 👍